### PR TITLE
TokenReview cache tests

### PR DESCRIPTION
**Pull request by jlpadilla**
Signed-off-by: Jorge Padilla <jpadilla@redhat.com>

**Related Issue:**  stolostron/backlog#22506

### Description of changes
- Adds test for TokenReview and cache.

### CODE REVIEW

File: pkg/rbac/tokenReview_test.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+	// Initialize cache state - pending TokenReview.
+	cache := newCache()
+	cache.tokenReviewsPending["1234567890-pending"] = []chan *tokenReviewResult{make(chan *tokenReviewResult)}
+
+	// Execute function
+	var testCH chan *tokenReviewResult
+	cache.doTokenReview(context.TODO(), "1234567890-pending", testCH)
+
+	// Validate result
+	if len(cache.tokenReviewsPending["1234567890-pending"]) != 2 {
```
Comment (Anxhela21) [inline pkg/rbac/tokenReview_test.go:106]: Why is the length 2 here? Aren't we only adding one pending token to tokenReviewsPending cache?

Comment (Anxhela21) [reply pkg/rbac/tokenReview_test.go:106]: Got it. Thank you.

File: pkg/rbac/tokenReview_test.go [hunk 2]
```diff
@@ -51,40 +64,45 @@ func Test_IsValidToken_usingCache(t *testing.T) {
...
+		t.Error("Expected token to be invalid.")
 	}
 	if err != nil {
 		t.Error("Received unexpected error from IsValidToken()", err)
 	}
-	// TODO: Verify that cache got updated.
-	if cache.tokenRevews["1234567890"].updatedAt
+	// Verify that cache was updated within the last 1 millisecond.
+	if cache.tokenReviews["1234567890"].updatedAt.Before(time.Now().Add(time.Duration(-1) * time.Millisecond)) {
+		t.Error("Expected the cached TokenReview to be updated within the last millisecond.")
```
Comment (Anxhela21) [inline pkg/rbac/tokenReview_test.go:90]: "within the last minute"

File: pkg/rbac/tokenReview_test.go [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+	return Cache{
+		// Use a fake Kubernetes authentication client.
+		authClient:          fake.NewSimpleClientset().AuthenticationV1(),
+		tokenReviews:        map[string]*tokenReviewResult{},
+		tokenReviewsPending: map[string][]chan *tokenReviewResult{},
+		tokenReviewsLock:    sync.Mutex{},
+	}
+}
+
+func Test_IsValidToken_emptyCache(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (jlpadilla) [reply pkg/rbac/tokenReview_test.go:26]: Done

File: pkg/rbac/tokenReview_test.go (hunk 2 above)
Comment (jlpadilla) [reply pkg/rbac/tokenReview_test.go:92]: Actually `millisecond` is correct here.  This verifies that the function found that the cache had expired and it updated it with new results.

**Pull request merged**